darwin/datatypes.py

```diff
@@ -32,6 +32,37 @@ def get_sub(self, annotation_type: str) -> Optional[SubAnnotation]:
                 return sub
 
 
+@dataclass(frozen=True, eq=True)
+class VideoAnnotation:
+    annotation_class: AnnotationClass
+    frames: Annotation
+    keyframes: List[bool]
+    segments: List[List[int]]
+    interpolated: bool
+
+    def get_frame(self, frame_index):
+        return frames[frame_index]
+
+    def get_data(self, only_keyframes=True, post_processing=None):
+        if not post_processing:
+            post_processing = id
+        return {
+            "frames": {
+                frame: {
+                    **post_processing(
+                        self.frames[frame],
```

Comment:
We use `self.frames[frame]` three times here, it's making things a bit difficult to read.

I'm not sure if I have any particular suggestion here, because we are using a dict comprehension and that's fine by me.

Maybe having a separate function that builds this dictionary would help